src/v/cluster/rm_stm.cc

```diff
@@ -1649,7 +1649,9 @@ model::offset rm_stm::last_stable_offset() {
     // We optimize for the case where there are no inflight transactional
     // batches to return the high water mark.
     if (unlikely(!_bootstrap_committed_offset)) {
-        return model::offset::min();
+        // return next offset to offset::min, to follow the policy that the LSO
```

Comment:
> Given the definition LSO is never negative.

Are we relying on this check in next_offset() to return offset{0}? If so just return offset{0} ?

```
  if (p < offset{0}) {
        return offset{0};
    }
```

Reply:
i just wanted to make it easier to read, i can return offset 0. i.e. next of not initialized is 0